storage/posix/antispam/badger.go

```diff
@@ -257,13 +256,34 @@ func (f *follower) Follow(ctx context.Context, lr tessera.LogReader) {
 
 				span.SetAttributes(followFromKey.Int64(otel.Clamp64(followFrom)))
 
-				if followFrom >= size {
+				if followFrom > logSize {
 					// Our view of the log is out of date, exit the busy loop and refresh it.
 					workDone = false
 					return nil
 				}
+				var err error
+				if followFrom >= logSize {
+					// Our view of the log is out of date, update it
+					logSize, err = lr.IntegratedSize(ctx)
+					if err != nil {
+						workDone = true
+						return fmt.Errorf("populate: IntegratedSize(): %v", err)
+					}
+					switch {
+					case followFrom > logSize:
```

Comment:
If you meant to keep the above, then this block will never be executed. If removing the block above works, it means that this condition is not covered by a test.